docs/files.md

```diff
@@ -0,0 +1,97 @@
+# Input and Output Files
+
+## constraints.txt
+
+The `constraints.txt` is the input file used to specify the exact version constraints of python packages that are needed to build a wheel. This file is similar to `requirements.txt` but instead of listing packages that must be installed, it sets constraints on versions of packages, ensuring that certain versions are either used or avoided during the wheel building process.
+
+Each line in the `constraints.txt` file should contain the name of a package followed by a version specifier. The version specifier can be one of the following:
+
+* `==`: Specifies an exact version of the package.
+* `<`, `<=`, `>`, `>=`: Specifies a version range.
+* `!=`: Excludes a specific version
```

Comment:
We could point to the upstream documentation for the syntax, too.

Reply:
Umm, which upstream documentation? The customization.md or README.md or general fromager documentation?

Reply:
That file format is described as part of the python packaging standards or pip documentation, I'm not sure which. Have a look through https://www.pypa.io/en/latest/specifications/ and look for the official pip docs to find a good reference.

Reply:
Okay, done!